### PR TITLE
Duplicate Macro Bind Indicator

### PR DESCRIPTION
**Pull request by JayMensink**
Issue #216

Solution: If you try to bind a macro to a key that is already bound, it will indicate this by making the bind key red.

### CODE REVIEW

File: src/main/java/tools/redstone/redstonetools/macros/Macro.java [hunk 1]
```diff
@@ -85,10 +86,12 @@ public void setKey(Key key) {
     }
 
     public void changeKeyBindingKeyCode() {
+
```
Comment (xtrm-en) [inline src/main/java/tools/redstone/redstonetools/macros/Macro.java:89]: why the spaces :sob:

**Review by xtrm-en — APPROVED** [review]
lgtm

File: src/main/java/tools/redstone/redstonetools/macros/Macro.java (hunk 1 above)
Comment (JayMensink) [reply src/main/java/tools/redstone/redstonetools/macros/Macro.java:89]: i was messing around and apparently i left these in on accident after adding and removing some code

File: src/main/java/tools/redstone/redstonetools/utils/KeyBindingUtils.java [hunk 2]
```diff
@@ -9,4 +11,12 @@ public static void removeKeyBinding(KeyBinding keyBinding) {
         ((KeyBindingMixin)keyBinding).removeKeybinding(keyBinding);
     }
 
+    public static boolean isKeyAlreadyBound(Key key) {
+        KeyBinding[] keyBindings = MinecraftClient.getInstance().options.allKeys;
+        for (KeyBinding keyBinding : keyBindings) {
+            if (keyBinding.getDefaultKey() == key) { return true; }
```
Comment (Matthias1590) [inline src/main/java/tools/redstone/redstonetools/utils/KeyBindingUtils.java:17]: Isn't the default key hardcoded and not the one set by the user?

Comment (JayMensink) [reply src/main/java/tools/redstone/redstonetools/utils/KeyBindingUtils.java:17]: ugh you're right time to fix that

Comment (JayMensink) [reply src/main/java/tools/redstone/redstonetools/utils/KeyBindingUtils.java]: Successfully changed.

**Review by Matthias1590 — APPROVED** [review]

**Review by Matthias1590 — APPROVED** [review]

**Comment by JayMensink** [comment]
@Matthias1590 can you merge this

**Pull request merged**